### PR TITLE
[CI] Build and Preview Test

### PR DESCRIPTION
**Pull request by leecalcote**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: .github/workflows/build-and-preview-site.yml [hunk 1]
```diff
@@ -19,8 +19,10 @@ jobs:
 
       - name: Deploy and Preview 🚀
         uses: JamesIves/github-pages-deploy-action@4.0.0
+        env:
+          PR_NUMBER: ${{ github.event.number }}
         with:
-          repository-name: "layer5labs/layer5"
+          repository-name: "layer5labs/layer5/{PR_NUMBER}"
```
Comment (Nikhil-Ladha) [inline .github/workflows/build-and-preview-site.yml:25]: That's an interesting way...🤔

Comment (Nikhil-Ladha) [reply .github/workflows/build-and-preview-site.yml:25]: Though, it has one drawback and that is what happens in case of multiple PR?

Comment (leecalcote) [reply .github/workflows/build-and-preview-site.yml:25]: I tossed out this suggestion a couple times amidst our conversations and am curious to know if we can be successful. It should be the case that we can. Here's the thought written down - https://github.com/layer5io/layer5/issues/1578.

Initial build works - https://layer5labs.github.io/layer5/

The Gatsby config will need to be tweaked.

Comment (leecalcote) [reply .github/workflows/build-and-preview-site.yml:25]: Multiple PRs = multiple deploys of the site in the preview repo, separated into different root folders (root according to that PR). There is some work to do to either dynamically update the Gatsby config during execution of this workflow.

Comment (Nikhil-Ladha) [reply .github/workflows/build-and-preview-site.yml:25]: That I understand, but the github preview will be live using any one of all the root folders. Which means, the last PR will be hosted or the latest deployment.
And, what if two PRs are pushed together?

**Comment by Nikhil-Ladha** [comment]
Clean is by default set to `true` you need to use `clean:false` to override it

File: .github/workflows/build-and-preview-site.yml (hunk 1 above)
Comment (leecalcote) [reply .github/workflows/build-and-preview-site.yml:25]: There's a solution in here somewhere. The PR # can be dynamically retrieved. A new folder with PR # as its name (guaranteed to be non-colliding) can be dynamically created (and deleted when PR is merged or closed).

Use of `--path-prefix` or other config might be necessary. The collection of built files and folders under `public` just need an http server to be served from, so stuffing copies of the site into individual folders and dropping a preview link into a comment on the PR at the end of the build-and-preview workflow should give us the DX we're looking for without restriction on minutes and $.

```
module.exports = {
  pathPrefix: `/yourpathprefix`,
}
```

**Comment by leecalcote** [comment]
Ah, I see. I'm going to shift my focus elsewhere for a couple days. Please do pickup this idea/pr/workflow and fiddle as you will.

**Comment by Nikhil-Ladha** [comment]
I don't think we have the test repo created?

**Comment by Nikhil-Ladha** [comment]
I am stuck with secrets, to use for updating the repo. Can you help me here?

**Comment by Nikhil-Ladha** [comment]
I think some permission needs to be updated,
`remote: Permission to layer5labs/layer5.git denied to l5io.`

**Comment by leecalcote** [comment]
@Nikhil-Ladha permissions updated...

**Comment by l5io** [comment]
This pull request is being automatically deployed to Netlify.

🔍 Inspect: https://app.netlify.com/sites/layer5ng/deploys/603a1134d68205aa17888474
✅ Preview: https://603a1134d68205aa17888474--layer5ng.netlify.app

**Review by Nikhil-Ladha — APPROVED** [review]

**Comment by Nikhil-Ladha** [comment]
Merging to test few things...

**Comment by l5io** [comment]
This pull request is being automatically deployed to Netlify.

🔍 Inspect: https://app.netlify.com/sites/layer5ng/deploys/603a150fbb172bb52b718c1d
✅ Preview: https://603a150fbb172bb52b718c1d--layer5ng.netlify.app

**Comment by l5io** [comment]
This pull request is being automatically deployed to Netlify.

🔍 Inspect: https://app.netlify.com/sites/layer5ng/deploys/603a1617060121b2f253681e
✅ Preview: https://603a1617060121b2f253681e--layer5ng.netlify.app

**Pull request merged**